src/components/floating-menu/floating-menu.js

```diff
@@ -175,11 +175,72 @@ class FloatingMenu extends mixin(createComponent, eventedShowHideState, trackBlu
       this._getContainer().appendChild(this.element);
       this._place();
       (this.element.querySelector(this.options.selectorPrimaryFocus) || this.element).focus();
+
+      // this.element.style.height = this.element.querySelectorAll(this.options.);
     }
     if (state === 'hidden' && this.hResize) {
       this.hResize.release();
       this.hResize = null;
     }
+
+    this.element.classList.toggle(classShown, shown);
+    if (classRefShown) {
+      refNode.classList.toggle(classRefShown, shown);
+    }
+
+    /**
+     *  system 360 motion
+     */
+    if(state === 'shown'){
+
+      //-----------------------------------------------------
+      //  test run to get the actual height
+      this.element.style.visibility = 'hidden';
+      this.element.style.height = 'auto';
+      let 
+        targetHeight = this.element.offsetHeight,
```

Comment:
This line causes [forced synchronous layout](https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing#avoid_forced_synchronous_layouts). It’d be great if we don’t have to get the dimension via JavaScript. Otherwise the code should run asynchronously e.g. in [rAF](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) callback.